marconi-sidechain/test/Spec/Marconi/Sidechain/RpcClientAction.hs

```diff
@@ -0,0 +1,74 @@
+module Spec.Marconi.Sidechain.RpcClientAction where
+
+import Control.Concurrent.STM (atomically)
+import Control.Lens ((^.))
+import Data.Proxy (Proxy (Proxy))
+import Marconi.ChainIndex.Indexers.Utxo qualified as Utxo
+import Marconi.Core.Storable qualified as Storable
+import Marconi.Sidechain.Api.Query.Indexers.Utxo qualified as UIQ
+import Marconi.Sidechain.Api.Routes (AddressUtxoResult, CurrentSyncedPointResult, JsonRpcAPI,
+                                     TxOutAtQuery (TxOutAtQuery))
+import Marconi.Sidechain.Api.Types (SidechainEnv, sidechainAddressUtxoIndexer, sidechainEnvIndexers)
+import Network.HTTP.Client (defaultManagerSettings, newManager)
+import Network.JsonRpc.Client.Types ()
+import Network.JsonRpc.Types (JsonRpcResponse)
+import Network.Wai.Handler.Warp (Port)
+import Network.Wai.Handler.Warp qualified as Warp
+import Servant.API ((:<|>) ((:<|>)))
+import Servant.Client (BaseUrl (BaseUrl), ClientEnv, HasClient (Client), Scheme (Http), client, hoistClient,
+                       mkClientEnv, runClientM)
+
+-- | Alias for the storable-action and RPC-client-Action pair, to simplify the type signatures
+data RpcClientAction = RpcClientAction
+    { insertUtxoEventsAction  :: !InsertUtxoEventsCallback
+    , queryAddressUtxosAction :: !(String -> IO (JsonRpcResponse String AddressUtxoResult))
+    , querySyncedPointAction  :: !(IO (JsonRpcResponse String CurrentSyncedPointResult))
+    }
+
+mkRpcClientAction :: SidechainEnv -> Port -> IO RpcClientAction
+mkRpcClientAction env port = do
+    manager <- newManager defaultManagerSettings
+    let clientEnv = mkClientEnv manager (baseUrl port)
+        (_ :<|> _ :<|> rpcSyncPoint :<|> rpcUtxos :<|> _ :<|> _ :<|> _)
+          = mkHoistedHttpRpcClient clientEnv
+    pure $ RpcClientAction (mkInsertUtxoEventsCallback env)
+                           (\a -> rpcUtxos $ TxOutAtQuery a Nothing)
+                           (rpcSyncPoint "")
+
+baseUrl :: Warp.Port -> BaseUrl
+baseUrl port = BaseUrl Http "localhost" port ""
+
+-- | A type for Storable Action to store events
```

Comment:
same as above. My error originally, but I think  `type`  should be changed to  `alias`